lib/rebay_api_web/plugs/authorize_user.ex

```diff
@@ -0,0 +1,22 @@
+defmodule RebayApiWeb.Plugs.AuthorizeUser do
+  import Plug.Conn
+  use RebayApiWeb, :controller
+
+  def init(_params) do
+  end
+
+  def call(conn, _params) do
+    user_uuid_param = conn.params["user_uuid"]
+    current_user = conn.assigns[:user]
+
+    if current_user != nil && user_uuid_param == current_user.uuid do
+      conn
+    else
+      conn
+      |> put_status(:unauthorized)
+      |> put_view(RebayApiWeb.ErrorView)
+      |> render("401.json")
```

Comment:
Should this be a 401, or a 403?

Reply:
I thought 401 because "unauthorized" sounded like exactly what we wanted, but it does seem to apply more specifically to authentication now that I've looked a little closer. 403 it is!